### ISSUE TITLE
h2 tag missing from product names on search results page

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#13089]
The change to [add an `h2` tag to the product names](https://github.com/SAP/spartacus/pull/11867/files#diff-6fc7f988e1705ccce3e4b94108854326fe98067f3a0cfd6eb5a4301593d00b15R21) in the search results page did not make it in during the epic branch merge to develop. 

Todo: add `h2` tag directly to `develop`

Related to:  #11151
